core-rust/p2p/src/column_families.rs

```diff
@@ -0,0 +1,50 @@
+use node_common::store::typed_cf_api::SborDbCodec;
+use node_common::store::typed_cf_api::{
+    DefaultCf, DirectDbCodec, UnitDbCodec,
+};
+use crate::address_book_components::AddressBookNodeId;
+use crate::migration::MigrationStatus;
+use crate::typed_cf_api::AddressBookNodeIdDbCodec;
+
+/// Address book and safety state store migration status. Filled once during the migration.
+pub struct MigrationStatusCf;
+impl DefaultCf for MigrationStatusCf {
+    type Key = ();
+    type Value = MigrationStatus;
+
+    const DEFAULT_NAME: &'static str = "migration_status";
+    type KeyCodec = UnitDbCodec;
+    type ValueCodec = SborDbCodec<MigrationStatus>;
+}
+
+/// Address book
+pub struct AddressBookCf;
+impl DefaultCf for AddressBookCf {
+    type Key = AddressBookNodeId;
+    type Value = Vec<u8>;
+
+    const DEFAULT_NAME: &'static str = "address_book";
+    type KeyCodec = AddressBookNodeIdDbCodec;
+    type ValueCodec = DirectDbCodec;
+}
+
+/// Safety store
+pub struct SafetyStoreCf;
+impl DefaultCf for SafetyStoreCf {
+    type Key = ();
+    type Value = Vec<u8>;
```

Comment:
I've just realised we're storing all these values as raw Vec<u8> instead of as upgradable SBOR. Why is that?

I'm concerned about this:
* DSON has built in support for adding new columns
* SBOR does not, so we need to add in something else to allow us to change our models over time. So far, our canonical choice has been wrapping things in a versioning enum. It's not ideal, but it's better than nothing

Just storing raw bytes without having the encoding/decoding models on the rust side also kinda defeats the purpose of moving the storage of this state into rust.

Could we tweak these to use `impl VersionedCf` instead?

Reply:
The current implementation does not send serialized `DSON` on the Rust side and does not rely on `DSON` capabilities. Instead, it sends SBOR-encoded `SafetyStateDTO`. This DTO and all other DTO's referred by `SafetyStateDTO` are a prototype for Rust-side implementation once the rest of the code will be moved to Rust. Currently, we don't need to decode this struct, so there is no need to implement all these DTO's on the Rust side. Part of the structs already present on the Rust side, but require some changes (like adding annotations). Other part is missing, but there are no relevant modules/infrastructure for the correct placement for these structs. Putting them all together looked impractical too.

Reply:
Well we can’t update these models in future unless they’re versioned.

So we’ll need to do a second migration unless we get this right now.

Reply:
Reworked to use versioned column families.